syntax/crisp_lexer.mll

```diff
@@ -95,6 +95,9 @@ let integer = ['0'-'9']+
 
 (*NOTE currently only Unix-style newline is supported, because it's simpler.*)
 let nl = '\n'
+(*NOTE we match nl with eof to maintain a correct line number in unix text files.
+  The whitespace and comment before it are to ensure it is the longest match. *)
+let end_of_file = nl* ' '* comment? nl* eof
```

Comment:
I wonder if the `end_of_file` set should be strengthened to include more combinations of the substrings it relies on, as in:
```
(nl|' '|comment)*eof
```